samples/snippets/fsharp/azure/table-storage.fsx

```diff
@@ -1,111 +1,90 @@
+#r "nuget:Azure.Data.Tables" // Load the Azure Data Tables nuget package
+
 open System
 open Azure
 open Azure.Data.Tables // Namespace for Table storage types
-open System.Linq
-open System.Collections.Generic
 
 //
 // Get your connection string.
 //
 
-let storageConnString = "..." // fill this in from your storage account
+let storageConnString = "UseDevelopmentStorage=true" // fill this in from your storage account
 
 //
 // Create the Table service client.
 //
 
 // Create the table client.
-let tableClient = TableServiceClient(storageConnString)
+let tableClient = TableServiceClient storageConnString
 
 //
 // Create a table.
 //
 
 // Retrieve a reference to the table.
-let table = tableClient.GetTableClient("people")
+let table = tableClient.GetTableClient "people"
 
 // Create the table if it doesn't exist.
-ignore(table.CreateIfNotExists())
+table.CreateIfNotExists() |> ignore
 
 //
 // Add an entity to a table. The last name is used as a partition key.
 //
 
 type Customer(firstName, lastName, email: string, phone: string) =
     interface ITableEntity with
-        member this.ETag
-            with get (): ETag = 
-                raise (System.NotImplementedException())
-            and set (v: ETag): unit = 
-                raise (System.NotImplementedException())
-        member this.PartitionKey
-            with get (): string = 
-                raise (System.NotImplementedException())
-            and set (v: string): unit = 
-                raise (System.NotImplementedException())
-        member this.RowKey
-            with get (): string = 
-                raise (System.NotImplementedException())
-            and set (v: string): unit = 
-                raise (System.NotImplementedException())
-        member this.Timestamp
-            with get (): Nullable<DateTimeOffset> = 
-                raise (System.NotImplementedException())
-            and set (v: Nullable<DateTimeOffset>): unit = 
-                raise (System.NotImplementedException())
+        member val ETag = ETag "" with get, set
```

Comment:
Would there be any value in having these delegate to the class-level implementation or vice-versa?

Reply:
Good question. I've never even done something like this before :-) It looks very much like if you have an interface and class member with the same name and type, F# seems to treat them as one and the same. I've not looked at the IL or decompiled it into C# to see. @cartermp can you confirm what happens there?

Reply:
In genera it's idiomatic F# to do it this way. I wouldn't try to implement it at the class level and then have the interface just call that.